apps/admin-server/src/pages/projects/[project]/widgets/resourcesmap/[id]/polygons.tsx

```diff
@@ -0,0 +1,146 @@
+import { useState } from 'react';
+import { Button } from '@/components/ui/button';
+import {
+  Form,
+  FormControl,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+} from '@/components/ui/form';
+import { Input } from '@/components/ui/input';
+import { Separator } from '@/components/ui/separator';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { useForm } from 'react-hook-form';
+import { Heading } from '@/components/ui/typography';
+import { useFieldDebounce } from '@/hooks/useFieldDebounce';
+import { EditFieldProps } from '@/lib/form-widget-helpers/EditFieldProps';
+import * as z from 'zod';
+import { ResourceOverviewMapWidgetTabProps } from '.';
+import useAreas from '@/hooks/use-areas';
+import { Checkbox } from '@/components/ui/checkbox';
+
+const formSchema = z.object({
+  customPolygon: z.array(z.object({ id: z.number(), name: z.string() })).optional(),
+  customPolygonUrl: z.any().optional()
+});
+
+export default function WidgetResourcesMapButton(
+  props: ResourceOverviewMapWidgetTabProps &
+    EditFieldProps<ResourceOverviewMapWidgetTabProps> & {
+      customPolygon?: any;
+      customPolygonUrl?: any;
+    }
+) {
+
+  type FormData = z.infer<typeof formSchema>;
+
+  async function onSubmit(values: FormData) {
+    const customPolygon = values?.customPolygon?.map((item: any) => {
+      const url = values?.customPolygonUrl[item.id];
```

Comment:
```suggestion
      const url = values?.customPolygonUrl[item.id] ?? "";
```

Zo krijgen we geen errors als die `null` of `undefined` is.